### ISSUE TITLE
Mis-aligned initialBlock cause problems

### ISSUE THREAD

**Issue by abourget** [streamingfast/substreams#262]
Multiple modules that have initialBlock that aren't aligned will cause problems. Seen on Solana, but applies to all Substreams engines.

---

Report from @Eduard-Voiculescu 

Currently there is an issue with the latest of develop when trying to run a substreams which pulls in an spkg and that the start block of the spkg is before the start block of the current substreams. (I do not know if this also happens if the start block is AFTER the current substreams).

How to reproduce:
```bash
substreams run hivemapper-v0.1.4.spkg map_outputs -e mainnet.sol.streamingfast.io:443 --production-mod
```

Expected error:
```bash
Error: rpc error: code = Unknown desc = rpc error: code = Internal desc = error during init_stores_and_backprocess: failed run_parallel_process: parallel processing run: scheduler run: process job result for module "map_outputs": job ended in error: receiving stream resp: rpc error: code = Internal desc = error building pipeline: failed to setup subrequest stores: load full store address_lookup_tables:store_address_lookup_tables_stage_2 (d27a4554b2125b91d3daf2847e09915ef09d751c): load full store address_lookup_tables:store_address_lookup_tables_stage_2 at 0158558168-0154656000.kv: opening file gs://dfuseio-global-substreams-uscentral/sol-mainnet/substreams-states/v4/d27a4554b2125b91d3daf2847e09915ef09d751c/states?project=dfuseio-global: not found
```

[hivemapper-v0.1.4.spkg.gz](https://github.com/streamingfast/substreams/files/12113650/hivemapper-v0.1.4.spkg.gz)

**Comment by maoueh** [comment]
#265 is also open, we need to close one @abourget @Eduard-Voiculescu, the one @Eduard-Voiculescu created as more meat I would keep his.

**Comment by sahra-karakoc** [comment]
I ran into a similar error due to multiple initialBlocks. 

Reproduce with:

```
substreams gui -e mainnet.eth.streamingfast.io:443 \
substreams.yaml \
graph_out \
--start-block 114468000
```

Error: 
```
Error: rpc error: code = Unknown desc = rpc error: code = Internal desc = error during init_stores_and_backprocess: run_parallel_process failed: parallel processing run: scheduler run: squash stage 0 module "store_graph_account_curator": loading partial: "0011447000-0011446769.964f88882e4871daf07e510c6c4c3083.partial": load partial store store_graph_account_curator at 0011447000-0011446769.964f88882e4871daf07e510c6c4c3083.partial: opening file: not found
```
[graph-network-substreams-v0.1.0.spkg.zip](https://github.com/streamingfast/substreams/files/12362172/graph-network-substreams-v0.1.0.spkg.zip)

**Comment by sduchesneau** [comment]
[Fixed in v1.6.0](https://github.com/streamingfast/substreams/releases/tag/v1.6.0)
